artifactory/services/upload.go

```diff
@@ -112,9 +112,10 @@ func addSymlinkProps(artifact clientutils.Artifact, uploadParams UploadParams) (
 		sha1Property := ""
 		fileInfo, err := os.Stat(artifact.LocalPath)
 		if err != nil {
-			return "", err
-		}
-		if !fileInfo.IsDir() {
+			if !os.IsNotExist(err) { // If error occurred, but not due to nonexistence of Symlink target -> return empty
+				return "", err
+			}
+		} else if !fileInfo.IsDir() { // If Symlink target exists -> get SHA1 if isn't a directory
 			file, err := os.Open(artifact.LocalPath)
 			errorutils.CheckError(err)
 			if err != nil {
```

Comment:
The above error check can be simplified to:
```
			if err != nil {
				return "", CheckError(err)
			}
```
and as a result, the following line can be removed:
```
			errorutils.CheckError(err)
```